core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/DatalakeV4Controller.java

```diff
@@ -28,11 +31,46 @@ public class DatalakeV4Controller implements DatalakeV4Endpoint {
     @Inject
     private WorkspaceService workspaceService;
 
+    @Inject
+    private DiskUpdateService diskUpdateService;
+
     @Override
+    @InternalOnly
     public StackViewV4Responses list(@Deprecated String environment, @TenantAwareParam String environmentCrn) {
         if (StringUtils.isBlank(environmentCrn) || StringUtils.isNotBlank(environment)) {
             throw new BadRequestException("environment param is deprecated, only environmentCrn should be filled.");
         }
         return stackOperations.listByEnvironmentCrn(workspaceService.getForCurrentUser().getId(), environmentCrn, List.of(StackType.DATALAKE));
     }
+
+    @Override
+    public Boolean isDiskTypeChangeSupported(String platform) {
+        if (StringUtils.isBlank(platform)) {
+            throw new BadRequestException("Platform param is null or empty.");
+        }
+        return diskUpdateService.isDiskTypeChangeSupported(platform);
```

Comment:
same here

Reply:
it is just validating the platform input which controllers do - so no

Reply:
you can use `@NotNull` annotation on parameter